src/connectors/azure_log.py

```diff
@@ -0,0 +1,333 @@
+        $1:properties.authenticationMethodsUsed::VARIANT, $1:properties.authenticationProcessingDetails::VARIANT,
+        $1:properties.clientAppUsed::STRING, $1:properties.conditionalAccessStatus::STRING, $1:properties.createdDateTime::TIMESTAMP_LTZ,
+        $1:properties.deviceDetail::VARIANT, $1:properties.id::STRING, $1:properties.ipAddress::STRING, $1:properties.isInteractive::BOOLEAN,
+        $1:properties.location::VARIANT, $1:properties.mfaDetail::VARIANT, $1:properties.networkLocationDetails::VARIANT,
+        $1:properties.processingTimeInMilliseconds::NUMBER, $1:properties.resourceDisplayName::STRING, $1:properties.resourceId::STRING,
+        $1:properties.riskDetail::STRING, $1:properties.riskEventTypes::VARIANT, $1:properties.riskLevelAggregated::STRING,
+        $1:properties.riskLevelDuringSignIn::STRING, $1:properties.riskState::VARIANT, $1:properties.status::VARIANT,
+        $1:properties.tokenIssuerType::STRING, $1:properties.userDisplayName::STRING, $1:properties.userId::STRING,
+        $1:properties.userPrincipalName::STRING, $1:resourceId::STRING, $1:resultDescription::STRING, $1:resultSignature::STRING,
+        $1:resultType::STRING, $1:tenantId::STRING, $1:time::TIMESTAMP_LTZ, CURRENT_TIMESTAMP()
+FROM @DATA.{name}_STAGE
+)
+"""
+    }
+
+    db.create_pipe(
+        name=f"{name}_PIPE",
+        sql=pipe_sql[options['log_type']],
+        replace=True)
+
+    db.execute(f'ALTER PIPE DATA.{name}_PIPE SET PIPE_EXECUTION_PAUSED=true')
+    db.execute(f'GRANT OWNERSHIP ON PIPE DATA.{name}_PIPE TO ROLE {SA_ROLE}')
+
+    return {'newStage': 'finalized', 'newMessage': 'Table, Stage, and Pipe created'}
+
+
+def ingest(name, options):
+    block_blob_service = BlockBlobService(
+        account_name=options['storage_account'],
+        sas_token=options['sas_token'],
+        endpoint_suffix=options['suffix']
+    )
+    base_name = name.rstrip('CONNECTION')
+
+    db.execute(f"select SYSTEM$PIPE_FORCE_RESUME('DATA.{base_name}_PIPE');")
+
+    files = block_blob_service.list_blobs(options['blob_name'])
+
+    timestamp_query = f"""
+    SELECT loaded_on
+    FROM data.{name}
+    ORDER BY loaded_on DESC
+    LIMIT 1
+    """
+    ts = next(db.fetch(timestamp_query), None)
+    last_loaded = ts['LOADED_ON'] if ts else None
+
+    log.info(f"Last loaded time is {last_loaded}")
+
+    new_files = []
+    if last_loaded:
+        for file in files:
+            if file.properties.creation_time > last_loaded:
+                new_files.append(StagedFile(file.name, None))
+    else:
+        for file in files:
+            new_files.append(StagedFile(file.name, None))
+
+    log.info(f"Found {len(new_files)} files to ingest")
+
+    # Proxy object that abstracts the Snowpipe REST API
+    ingest_manager = SimpleIngestManager(account=environ.get('SNOWFLAKE_ACCOUNT'),
+                                         host=f'{environ.get("SNOWFLAKE_ACCOUNT")}.snowflakecomputing.com',
```

Comment:
ditto — if `environ.get('SNOWFLAKE_ACCOUNT')` is `None`, we want to error before we try anything else

Reply:
resolved